independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java

```diff
@@ -21,7 +23,7 @@
  * The implementation also contains optimizations that allow the ClassLoader to keep a minimum number of jars open
  * while also preventing the lookup of the entire classpath for missing resources in known directories (like META-INF/services).
  */
-public final class RunnerClassLoader extends ClassLoader {
```

Comment:
must it implement the interface - it is not sufficient just having methods on the class ?

Reply:
Yes, an object should implement the interface to be able to register. If this is a problem (RunnerClassLoader implementing Resource interface), I think it should be possible to create an inner class implementing the interface and keep the RunnerClassLoader signature unchanged. Should I change the code this way?

Reply:
Yeah; I think the less exposure we have of crac api the better as it (as I see it) just a internal implementation detail. 

Ultimately if we could find a way to have this as an extension so it only "cost" when crac is possible to enable.  

Also...does this build and run in native ? (Ie. I know crac won't be enabled but would need it to not prevent use of native - at least if available in core)

Reply:
I wouldn't worry about crac building in native, we can hint it to be seen as dead code during analysis.

Reply:
Sure, I'm changing this piece.

I've tried building my quarkus-example in native mode without any issue. This surprised me a bit :) (org-crac uses reflection to access API), but it went smoothly. Also, I'm relying on GHA to run test in native as well.